tests/containers/volumes.pm

```diff
@@ -0,0 +1,125 @@
+    my $ret = script_run("$runtime run --rm --volumes-from $test_container $test_image ls /$test_dir/$test_file");
+    if ($ret != 0) {
+        if ($runtime eq "podman") {
+            record_soft_failure("gh#19529 - Unexpected error with --volumes-from") if ($ret != 0 && $runtime == "podman");
+        } else {
+            die("--volumes-from failed");
+        }
+    }
+
+    assert_script_run("$runtime rm -vf $test_container");
+
+    # Test --volume option with directory (read-only)
+    assert_script_run("! $runtime run --rm --volume \$PWD/$test_dir:/$test_dir:ro,Z $test_image rm /$test_dir/$test_file");
+    assert_script_run("test -f $test_dir/$test_file");
+
+    # Equivalent --mount option to above
+    assert_script_run("! $runtime run --rm --mount type=bind,source=\$PWD/$test_dir,destination=/$test_dir,readonly$Z $test_image rm /$test_dir/$test_file");
+    assert_script_run("test -f $test_dir/$test_file");
+
+    assert_script_run("rm $test_dir/$test_file");
+
+    # Test --volume option with directory (read-write)
+    assert_script_run("$runtime run --rm --volume \$PWD/$test_dir:/$test_dir:Z $test_image touch /$test_dir/$test_file");
+    assert_script_run("test -f $test_dir/$test_file");
+
+    assert_script_run("rm $test_dir/$test_file");
+
+    # Equivalent --mount option to above
+    assert_script_run("$runtime run --rm --mount type=bind,source=\$PWD/$test_dir,destination=/${test_dir}$Z $test_image touch /$test_dir/$test_file");
+    assert_script_run("test -f $test_dir/$test_file");
+
+    # Test volume subcommands
+    assert_script_run("$runtime volume create $test_volume");
+    assert_script_run("$runtime volume ls --format '{{.Name}}' | grep -Fx $test_volume");
+    assert_script_run("$runtime volume inspect --format '{{.Name}}' $test_volume | grep -Fx $test_volume");
+    # Other subcommands supported only by podman: mount unmount reload
+    if ($runtime eq "podman") {
+        assert_script_run("$runtime volume exists $test_volume");
+        assert_script_run("tar cf - $test_dir | $runtime volume import $test_volume -");
+        assert_script_run("$runtime volume export $test_volume | tar tf - | grep -Fx $test_dir/$test_file");
+    }
+    assert_script_run("$runtime volume rm $test_volume");
+
+    for (my $i = 1; $i <= 10; $i++) {
+        assert_script_run("$runtime volume create $test_volume");
+
+        # Test --volume option with volume (read-write)
+        assert_script_run("$runtime run --rm --volume $test_volume:/$test_dir $test_image touch /$test_dir/$test_file");
+        assert_script_run("test -f $test_dir/$test_file");
+
+        # Equivalent --mount option to above
+        # NOTE: ",Z" in --mount is known to work on 4.6.0+ but not 4.4.4
+        my $podman_version = get_podman_version();
+        my $ZZ = ($runtime eq "podman" && package_version_cmp($podman_version, '4.6.0') < 0) ? "" : $Z;
```

Comment:
```suggestion
        my $conditionalLabel = ($runtime eq "podman" && package_version_cmp($podman_version, '4.6.0') < 0) ? "" : $Z;
```
Not a political statement, but Z and then ZZ, which is Z but only under certain conditions ... Yes it works, but it is confusing.

Let's not be lazy in naming things :stuck_out_tongue:

Reply:
> Not a political statement, but Z and then ZZ, which is Z but only under certain conditions ... Yes it works, but it is confusing.
> 
> Let's not be lazy in naming things 😛

I was thinking of ZZTop...